src/core/mantle/evaluator/simple/expression/expression.ts

```diff
@@ -24,11 +25,23 @@ export class Expression {
     this._expression = expression;
     this._feature = feature;
     let literalJP: JPLiteral[] = [];
-    expression = replaceDefines(expression, defines);
-    [expression, literalJP] = replaceVariables(
-      removeBackslashes(expression),
-      this._feature?.properties,
-    );
+
+    const hasJSONPath = /\$[a-zA-Z[.]/.test(expression);
```

Comment:
```suggestion
    const hasJSONPath = /^\$([.](?!$)|\[[^\]]*])*$/.test(expression);
```

Reply:
The above regex will not be check statements like `$..book[?(@.price<10)]` because it contains the `..` operator and a filter expression `?(@.price<10)`

Reply:
I think we don't need to check entire expression because checking entire expression needs some extra cost like needing extra regexp steps.
So I think we can only check first two word.
JSON Path shold have ‘$‘ as first word. And second should be ‘.‘ or a-zA-Z.

Reply:
I see. You can merge this PR then.
Keep in mind that this simpler pattern may produce false positives, so you may need to further validate the matched strings to ensure that they are valid JSONPath expressions if that is a requirement for your use case.

Reply:
I rethought it based on your suggestion, I found a more efficient way. Could you check this commit?
[fix: use variable prefix](https://github.com/reearth/reearth-web/pull/574/commits/6872ec3ecb567f1a5a0cf9c3ff2a7798449b1297)